From 531dede1021098b5c9598f734e8994060eca2b1e Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 24 Dec 2014 16:51:22 +0800
Subject: [PATCH 1130/1543] MLK-10055-2: mfd: si476x-i2c: sound is registered
 when no FM module attached

commit f3f54ecbe227e9ce054671bee1d7c6595e76af79 from
git://git.freescale.com/imx/linux-2.6-imx.git

The si476x_core_get_revision_info will send i2c command to FM module, if it
return error, there is no FM modules attached, so we need't to register the
sound card. otherwise, the pulseaudio will access this sound card, but return
a lot of i2c error.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 drivers/mfd/si476x-i2c.c |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/mfd/si476x-i2c.c b/drivers/mfd/si476x-i2c.c
index d9cc07a..2ca61bf 100644
--- a/drivers/mfd/si476x-i2c.c
+++ b/drivers/mfd/si476x-i2c.c
@@ -810,15 +810,14 @@ static int si476x_core_probe(struct i2c_client *client,
 	core->power_state = SI476X_POWER_UP_FULL;
 	si476x_core_set_power_state(core, SI476X_POWER_DOWN);
 
+	rval = si476x_core_get_revision_info(core);
+	if (rval < 0) {
+		rval = -ENODEV;
+		goto free_kfifo;
+	}
+
 	if (of_property_read_bool(client->dev.of_node, "revision-a10"))
 		core->revision = SI476X_REVISION_A10;
-	else {
-		rval = si476x_core_get_revision_info(core);
-		if (rval < 0) {
-			rval = -ENODEV;
-			goto free_kfifo;
-		}
-	}
 
 	cell_num = 0;
 
-- 
1.7.5.4

